jax/numpy/lax_numpy.py

```diff
@@ -92,15 +92,21 @@ class ndarray(six.with_metaclass(_ArrayMeta, onp.ndarray)):
 size = onp.size
 _dtype = lax._dtype
 
-bool_ = onp.bool_
-uint8 = onp.uint8
-uint16 = onp.uint16
-uint32 = onp.uint32
-uint64 = onp.uint64
-int8 = onp.int8
-int16 = onp.int16
-int32 = onp.int32
-int64 = onp.int64
+def _wrap_intdtype(cls):
+  class IntDtypeSubclass(cls):
+    def __new__(_, x):
+      return lax.convert_element_type(x, cls)
+  return IntDtypeSubclass
+
+bool_ = _wrap_intdtype(onp.bool_)
+uint8 = _wrap_intdtype(onp.uint8)
+uint16 = _wrap_intdtype(onp.uint16)
+uint32 = _wrap_intdtype(onp.uint32)
+uint64 = _wrap_intdtype(onp.uint64)
+int8 = _wrap_intdtype(onp.int8)
+int16 = _wrap_intdtype(onp.int16)
+int32 = _wrap_intdtype(onp.int32)
+int64 = _wrap_intdtype(onp.int64)
 float16 = onp.float16
```

Comment:
Do you plan to do the float/complex types, too? I would have expected them to have the same issue.

Reply:
You’re right, I’ll do that.